src/TooltipPopoverWrapper.js

```diff
@@ -231,54 +250,55 @@ class TooltipPopoverWrapper extends React.Component {
           document.addEventListener('click', this.handleDocumentClick, true);
         }
 
-        if (this._target) {
+        if (this._targets && this._targets.length) {
           if (triggers.indexOf('hover') > -1) {
-            this._target.addEventListener(
+            this.addEventOnTargets(
               'mouseover',
               this.showWithDelay,
               true
             );
-            this._target.addEventListener(
+            this.addEventOnTargets(
               'mouseout',
               this.hideWithDelay,
               true
             );
           }
           if (triggers.indexOf('focus') > -1) {
-            this._target.addEventListener('focusin', this.show, true);
-            this._target.addEventListener('focusout', this.hide, true);
+            this.addEventOnTargets('focusin', this.show, true);
+            this.addEventOnTargets('focusout', this.hide, true);
           }
-          this._target.addEventListener('keydown', this.onEscKeyDown, true);
+          this.addEventOnTargets('keydown', this.onEscKeyDown, true);
         }
       }
     }
   }
 
   removeTargetEvents() {
-    if (this._target) {
-      this._target.removeEventListener(
+    if (this._targets) {
+      this.removeEventOnTargets(
         'mouseover',
         this.showWithDelay,
         true
       );
-      this._target.removeEventListener(
+      this.removeEventOnTargets(
         'mouseout',
         this.hideWithDelay,
         true
       );
-      this._target.removeEventListener('keydown', this.onEscKeyDown, true);
-      this._target.removeEventListener('focusin', this.show, true);
-      this._target.removeEventListener('focusout', this.hide, true);
+      this.removeEventOnTargets('keydown', this.onEscKeyDown, true);
+      this.removeEventOnTargets('focusin', this.show, true);
+      this.removeEventOnTargets('focusout', this.hide, true);
     }
 
     document.removeEventListener('click', this.handleDocumentClick, true)
   }
 
   updateTarget() {
-    const newTarget = getTarget(this.props.target);
-    if (newTarget !== this._target) {
+    const newTarget = getTarget(this.props.target, true);
+    if (newTarget !== this._targets) {
```

Comment:
Using reference equality here to compare two arrays doesn't look right

Reply:
Yes, it seems like this would always be `true` since we would be getting a new array from `getTarget` every time. Probably need to perform a deep comparison here.

Reply:
And then only remove the event listeners from the targets that actually got removed (vs removing from all the targets if one was removed)

Reply:
it might be easier _and quicker_ to just remove them all and add the newTarget back 🤷‍♂

Reply:
True